--- eclips_Dec2017.f.orig	2018-09-21 17:49:55.007789083 +0200
+++ eclips_Dec2017.f	2018-09-21 17:50:36.415529470 +0200
@@ -115,6 +115,8 @@
 C                                    FOR SOLVING SOME TRIANGLES (Luc Maisonobe)
 C                      Jul 10, 2018  TAKE ECCENTRICITY INTO ACCOUNT WHEN
 C                                    COMPUTING ORBITAL ANGULAR RATE (Luc Maisonobe)
+C                      Jul 10, 2018  ENSURE CONVERGENCE OF GLONASS YAWEND
+C                                    ITERATIVE COMPUTATION (Luc Maisonobe)
 C                      Sep 14, 2018  FIXED BODY-X UNIT VECTOR COMPUTATION
 C                                    EXACTLY AT NODE (AVOID NaNs), IMPROVE
 C                                    ITS ACCURACY FOR NON-PERFECTLY CIRCULAR
@@ -370,12 +372,15 @@
 C GLONASS NOON TURN MODE ACORDING TO DILSSNER 2010 
          YAWEND=75.D0
 C  ITERATION FOR YAWEND OF THE GLONASS  NOON TURN
-         DO J=1,3
-           YAWEND=ABS(ATAN2(-TAN(BETADG*DTR),SIN(PI-
-     &       DTR*MURATE*YAWEND/YRATE(IPRN)))/DTR -      
-     &       ATAN2(-TAN(BETADG*DTR),SIN(PI+
-     &       DTR*MURATE*YAWEND/YRATE(IPRN)))/DTR)/2.D0 
-         END DO
+          J=0
+ 10       CONTINUE
+            J=J+1
+            TMP=YAWEND
+            YAWEND=ABS(ATAN2(-TAN(BETADG*DTR),SIN(PI-
+     &             DTR*MURATE*YAWEND/YRATE(IPRN)))/DTR -      
+     &             ATAN2(-TAN(BETADG*DTR),SIN(PI+
+     &             DTR*MURATE*YAWEND/YRATE(IPRN)))/DTR)/2.D0 
+          IF (J.LT.50 .AND. ABS(YAWEND-TMP).GT.1.0D-8) GOTO 10
 C UPDATE ANOON, CNOON FOR NEW GLONASS NOON TURN LIMITS
           ANOON= MURATE*YAWEND/YRATE(IPRN)
           CNOON= DCOS(ANOON*DTR)
