--- eclips_May2017.f	2018-02-08 18:05:06.479976742 +0100
+++ eclips_May2017_patched.f	2018-02-08 18:11:18.234201454 +0100
@@ -171,6 +171,7 @@
       REAL*8    ECLETM(MAXSAT,*)
       REAL*8    ANOON, ANIGHT
       REAL*8    CNOON, CNIGHT
+      REAL*8    C, S, QX, QY, QZ
       REAL*8    DTR, DTTAG
       REAL*8    XSV(*), SANTXYZ(*), VSVC(*), BETA, MURATE, YANGLE, DET,
 C Jan 10, 2017
@@ -715,17 +716,15 @@
            END IF
 C ROTATE X-VECTOR TO ECLIPSING YAW ANGLE PHI 
 C ECLIPSING (II/IIA) NOT TO BE USED  A HALF HR AFTER SHADOW !
-       SANTX=(COS((PHI-YANGLE)*DTR)*(V(2)-V(3)*R(2)/R(3))-COS(PHI*
-     & DTR)*
-     &(SANTXYZ(2)-SANTXYZ(3)*R(2)/R(3)))/(SANTXYZ(1)*V(2)-SANTXYZ(2)*v(1
-     &)+((SANTXYZ(2)*V(3)-SANTXYZ(3)*V(2))*R(1)+(SANTXYZ(3)*V(1)-SANTXYZ
-     &(1)*V(3))*R(2))/R(3))
-       SANTY = (COS(PHI*DTR) - (V(1)-V(3)*R(1)/R(3))*SANTX)/
-     & (V(2)-V(3)*R(2)/R(3))
+          C  = COS((PHI-YANGLE)*DTR)
+          S  = SIN((PHI-YANGLE)*DTR)
+          QX = C*SANTXYZ(1)+S*(R(2)*SANTXYZ(3)-R(3)*SANTXYZ(2))
+          QY = C*SANTXYZ(2)+S*(R(3)*SANTXYZ(1)-R(1)*SANTXYZ(3))
+          QZ = C*SANTXYZ(3)+S*(R(1)*SANTXYZ(2)-R(2)*SANTXYZ(1))
 C THE BODY-X UNIT VECTOR ROTATED BY (PHI-YANGLE) RETURNED
-          SANTXYZ(1)= SANTX
-          SANTXYZ(2)= SANTY
-          SANTXYZ(3)= (-R(1)*SANTX-R(2)*SANTY)/R(3)
+          SANTXYZ(1)= QX
+          SANTXYZ(2)= QY
+          SANTXYZ(3)= QZ
       END IF
       ENDIF
 1     RETURN
