--- eclips_May2017.f	2018-07-10 16:47:31.435181081 +0200
+++ eclips_May2017_patched.f	2018-07-10 16:48:59.919178494 +0200
@@ -107,6 +107,8 @@
 C                                    OF GPS TURN RECOVERY (Luc Maisonobe)
 C                      Jul 10, 2018  TAKE ECCENTRICITY INTO ACCOUNT WHEN
 C                                    COMPUTING ORBITAL ANGULAR RATE (Luc Maisonobe)
+C                      Jul 10, 2018  FIXED MISSING COMPUTATION OF BODY-X VECTOR
+C                                    AT END OF BLOCK-IIR TURN (Luc Maisonobe)
 C
 C     PARAMETERS        DESCRIPTION
 C        IDIR           DIRECTION OF PROCESSING (1=FORWARD, -1=BACKWARD)
@@ -645,7 +647,7 @@
      &      +SIGN(YRATE(IPRN),BETADG)*(TTAG-ECLSTM(IPRN,I)) 
 C Jan 16, 2015
 C               IF((PHI/YANGLE).GE.1.d0.OR.(PHI/YANGLE).LT.0.d0) GO TO 1
-                IF((YANGLE/PHI).GE.1.d0.OR.(PHI/YANGLE).LT.0.d0) GO TO 1
+                IF((YANGLE/PHI).LT.1.d0) PHI = YANGLE
                END IF
 C jan 10, 2017 - GALILEO night - start
                IF(IPRN.GT.64.AND.IPRN.LE.100) THEN
