--- eclips_Dec2017.f.orig	2018-09-27 17:17:04.673121101 +0200
+++ eclips_Dec2017.f	2018-09-27 17:30:29.725097562 +0200
@@ -110,6 +110,9 @@
 C                      Jul 09, 2018  FIXED BODY-X UNIT VECTOR FOR BEIDOU,
 C                                    CONSIDERING NON-PERFECTLY CIRCULAR
 C                                    ORBITS (Luc Maisonobe)
+C                      Jul 09, 2018  IMPROVED ACCURACY BY REPLACING PROJECTED
+C                                    PLANAR GEOMETRY WITH SPHERICAL TRIGONOMETRY
+C                                    FOR SOLVING SOME TRIANGLES (Luc Maisonobe)
 C
 C     PARAMETERS        DESCRIPTION
 C        IDIR           DIRECTION OF PROCESSING (1=FORWARD, -1=BACKWARD)
@@ -221,10 +224,10 @@
 C    &          YRATE(64), BETADG, PHI, SANTX, SANTY,        v(3),r(3)
      &          YRATE(136),BETADG, PHI, SANTX, SANTY, v(3),r(3), BETA0,
 C    &          SMTH, BETAE
-     &          BETAE, P2, V2, PV, COEFFP, COEFFV
+     &          BETAE, P2, V2, PV, COEFFP, COEFFV, TMP
       REAL*8    YAWEND
       REAL*8    SQRT, ACOS, ATAN, DCOS, COS, ATAN2, SIN, TAN
-      REAL*8    DABS, ABS, SIGN, DMOD, MAX
+      REAL*8    DABS, ABS, SIGN, DMOD, MAX, MIN
 C Dec 12, 2013
       REAL*8 YBIAS
       INTEGER*4 IBLK(*), J, I
@@ -420,8 +423,8 @@
           BETAINI(IPRN)= BETADG
           ECLSTM(IPRN,1) = TTAG
        ENDIF
-C DET - APPROX ORBIT ANGLE
-       DET= 180.0D0-ACOS(SVBCOS)/DTR
+C DET - ORBIT ANGLE
+       DET= 180.0D0-ACOS(SVBCOS / COS(BETADG*DTR))/DTR
        IF(ABS(YANGLE).GT.90.0D0) DET= 360.0D0 -DET
 C May 03, 2017 THE ON LIMIT BETADG <= 4 DEG
 C  MEO YS/ON SWITCHING DELTA BETA LIMIT
@@ -530,7 +533,7 @@
       END IF
 C Jan 10, 2017  end
       IF ( (NIGHT .OR. NOON)) THEN
-       DET=SQRT(MAX(0.0D0,(180.d0-acos(svbcos)/DTR)**2-BETADG**2))
+       DET=ACOS(MAX(-1.0D0, MIN(1.0D0, -SVBCOS / COS(BETADG*DTR))))/DTR
        PHI = PI/2.d0/DTR
 C Check if already after a midnight or noon
        IF(NIGHT) THEN
@@ -546,7 +549,7 @@
 C Jan 16, 2015- end
        END IF 
        IF(NOON) THEN
-       DET=SQRT(MAX(0.0D0,(acos(svbcos)*180.0D0/pi)**2-BETADG**2))
+       DET=ACOS(MAX(-1.0D0,MIN(1.0D0, SVBCOS / COS(BETADG*DTR))))/DTR
 C Jan 16, 2015- start (NO IIR X-REVERSAL!)
 c        IF(IBLK(IPRN).EQ.4.OR.IBLK(IPRN).EQ.5) THEN
 c         IF(DABS(YANGLE).LT.90.d0) DET=-DET
@@ -601,10 +604,12 @@
 C    &         .OR.(IPRN .GT.64.AND.IPRN.LE.100)).AND.NIGHT) THEN
      &     .OR.(IPRN .GT.64.AND.IPRN.LE.100).OR.IPRN.EQ.INPRN)
      &     .AND.NIGHT) THEN
+           TMP = ACOS(MAX(-1.0D0, MIN(1.0D0,
+     &                -COS(ANIGHT*DTR) / COS(BETADG*DTR)))) / DTR
            ECLSTM(IPRN,NECLIPS(IPRN))= ECLSTM(IPRN,NECLIPS(IPRN))-
-     &      SQRT(MAX(0.0D0,(ANIGHT-180.d0)**2-BETADG**2))/MURATE
+     &      TMP/MURATE
            ECLETM(IPRN,NECLIPS(IPRN))= ECLSTM(IPRN,NECLIPS(IPRN))+
-     &      2.d0*SQRT(MAX(0.0D0,(ANIGHT-180.d0)**2-BETADG**2))/MURATE
+     &      2.d0*TMP/MURATE
           END IF
       END IF
 C
@@ -668,10 +673,12 @@
 C    &           .OR.(IPRN .GT.64.AND.IPRN.LE.100)).AND.NIGHT) THEN
      &       .OR.(IPRN .GT.64.AND.IPRN.LE.100).OR.IPRN.EQ.INPRN)
      &       .AND.NIGHT) THEN
+             TMP = ACOS(MAX(-1.0D0, MIN(1.0D0,
+     &                  -COS(ANIGHT*DTR) / COS(BETADG*DTR)))) / DTR
              ECLSTM(IPRN,NECLIPS(IPRN))= ECLSTM(IPRN,NECLIPS(IPRN))-
-     &       SQRT(MAX(0.0D0,(ANIGHT-180.d0)**2-BETADG**2))/MURATE
+     &       TMP/MURATE
              ECLETM(IPRN,NECLIPS(IPRN))= ECLSTM(IPRN,NECLIPS(IPRN))+
-     &       2.d0*SQRT(MAX(0.0D0,(ANIGHT-180.d0)**2-BETADG**2))/MURATE
+     &       2.d0*TMP/MURATE
             END IF
         END IF
         ENDIF
