--- eclips_Dec2017.f.orig	2018-09-20 14:49:44.284396704 +0200
+++ eclips_Dec2017.f	2018-09-20 14:50:01.520397137 +0200
@@ -113,6 +113,8 @@
 C                      Jul 09, 2018  IMPROVED ACCURACY BY REPLACING PROJECTED
 C                                    PLANAR GEOMETRY WITH SPHERICAL TRIGONOMETRY
 C                                    FOR SOLVING SOME TRIANGLES (Luc Maisonobe)
+C                      Jul 10, 2018  FIXED OVERSHOOT OF NOMINAL YAW AT THE END
+C                                    OF GPS TURN RECOVERY (Luc Maisonobe)
 C
 C     PARAMETERS        DESCRIPTION
 C        IDIR           DIRECTION OF PROCESSING (1=FORWARD, -1=BACKWARD)
@@ -772,7 +774,10 @@
                    IF(ABS(SANTX).GT.180.D0) SANTX = SANTX -360.D0*
      &               SANTX /ABS(SANTX )
 C STOP! THE NOMINAL YAW (YANGLE) REACHED!
-                  IF(ABS(SANTX).GT.ABS(YAWEND)) GOTO 1
+                  IF(ABS(SANTX).GT.ABS(YAWEND)) THEN
+                      PHI = YANGLE
+                      GOTO 1
+                  ENDIF
                   IF(YAWEND.NE.0.D0.AND.((SANTX)/YAWEND).LT.0.D0) GOTO 1
 C SET PHI <-180,+180>
                    PHI= DMOD(PHI, 360.D0)
